src/main/java/io/github/thebusybiscuit/slimefun4/core/networks/energy/EnergyNet.java

```diff
@@ -244,7 +249,7 @@ private int tickAllGenerators(@Nonnull LongConsumer timings) {
 
             try {
                 Config data = BlockStorage.getLocationInfo(loc);
-                int energy = provider.getGeneratedOutput(loc, data);
+                int energy = provider.getGeneratedOutput(loc, data) / Math.max(1, (int) Slimefun.getNetworkManager().getNetworksFromLocation(loc, EnergyNet.class).stream().filter(Predicate.not(EnergyNet::isSaturated)).count());
```

Comment:
1. Don't use Streams
2. Keep in mind this code will run A LOT so performance will suffer from this addition.

Reply:
I refactored the stream to a for loop. Is it better now?